src/components/ChecklistForm.tsx

```diff
@@ -0,0 +1,797 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { ChecklistFormProps } from '@/types/todo';
+import { createDefaultTodo, calculateTodoComplexity, calculateChecklistProgress } from '@/utils/todoUtils';
+
+export default function ChecklistForm({ onAddChecklist, parentTodoId }: ChecklistFormProps) {
+  const [title, setTitle] = useState('');
+  const [items, setItems] = useState<string[]>([]);
+  const [newItem, setNewItem] = useState('');
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [showAdvanced, setShowAdvanced] = useState(false);
+  const [validationErrors, setValidationErrors] = useState<string[]>([]);
+  const [charCount, setCharCount] = useState(0);
+  const [estimatedTime, setEstimatedTime] = useState(60);
```

Comment:
[MEDIUM] Hardcoded estimated time value 30 used without constant.

**Code Smell** detected

Suggested change:

```suggestion
const DEFAULT_ESTIMATED_TIME = 30;
const [estimatedTime, setEstimatedTime] = useState(DEFAULT_ESTIMATED_TIME);
```